token-swap/program/src/curve.rs

```diff
@@ -27,14 +27,19 @@ impl SwapResult {
         fee_denominator: u64,
     ) -> Option<SwapResult> {
         let invariant = swap_source_amount.checked_mul(swap_destination_amount)?;
-        let new_source_amount = swap_source_amount.checked_add(source_amount)?;
-        let new_destination_amount = invariant.checked_div(new_source_amount)?;
-        let remove = swap_destination_amount.checked_sub(new_destination_amount)?;
-        let fee = remove
+
+        // assess the fee to calculate the amount swapped
+        let fee = source_amount
             .checked_mul(fee_numerator)?
             .checked_div(fee_denominator)?;
-        let new_destination_amount = new_destination_amount.checked_add(fee)?;
-        let amount_swapped = remove.checked_sub(fee)?;
+        let new_source_amount = swap_source_amount
```

Comment:
```suggestion
        let new_source_amount_less_fee = swap_source_amount
```
...or similar distinct variable name makes the math (and policy) a little clearer here. Will make it easier to write up docs (something like the link in the issue?) later on.